Microsoft.Toolkit.Uwp.UI.Controls/MasterDetailsView/MasterDetailsView.cs

```diff
@@ -250,15 +231,19 @@ private void OnBackRequested(object sender, BackRequestedEventArgs args)
         {
             if (ViewState == MasterDetailsViewState.Details)
             {
-                SelectedItem = null;
+                // let the OnFrameNavigating method handle it if
+                if (_frame == null || !_frame.CanGoBack)
```

Comment:
I don't understand this change. We'll always mark as handled, but only set the SelectedItem to null if the frame is null or if the frame cannot go back. So if the frame can go back, we mark this as handled but do not set the selected item to null?